django/db/backends/mysql/base.py

```diff
@@ -272,7 +273,7 @@ def init_connection_state(self):
             with self.cursor() as cursor:
                 cursor.execute('SET ' + ', '.join(assignments))
 
-    def create_cursor(self, name=None):
+    def create_cursor(self, name=None, chunk_size=constants.GET_ITERATOR_CHUNK_SIZE):
```

Comment:
I prefer a numerical value over `None` because it gives hints on the expected parameter type. Also, it's an error for both PostgreSQL and Oracle to provide `None` for respectively `itersize` and `arraysize`.

If we choose to set the default value for `chunk_size` to `None`, I think most developers will assume that providing `None` let the SQL adapter use its default value, which is not the case for the two adapters implementing server-side cursors.

I am not sure that `GET_ITERATOR_CHUNK_SIZE` is the right constant for the server-side cursor's chunk size. I think 100 is too low to be a good default value (since [apollo13's comment on the server-side cursors' PR](https://github.com/django/django/pull/7587#issuecomment-268900072)) and I'm in favor of creating a new constant for this default, maybe: `DEFAULT_CHUNK_SIZE`.
If you think that a new constant is the way to go, I'll write to the mailing list about this constant's value. IIUC, cx_Oracle retrieves results using server-side cursors for every query, so Oracle users will be strongly affected by this choice.